python/cudf/cudf/core/reshape.py

```diff
@@ -215,7 +230,30 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
     # when axis is 1 (column) we can concat with Series and Dataframes
     if axis == 1:
 
+        if join == "inner" and typ is cudf.Series:
+            new_df = cudf.concat(objs, axis=1)
+            indexes_list = []
+            for obj in objs:
+                indexes_list.append(cp.array(obj.index))
```

Comment:
This isn't safe as `obj.index` could be a string type or a MultiIndex for example.